DDDEastAnglia/Areas/Admin/Controllers/SponsorController.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Drawing;
+using System.Drawing.Imaging;
+using System.IO;
+using System.Linq;
+using System.Web;
+using System.Web.Mvc;
+using DDDEastAnglia.DataAccess;
+using DDDEastAnglia.DataAccess.SimpleData.Models;
+using DDDEastAnglia.Helpers;
+using SponsorModel = DDDEastAnglia.Areas.Admin.Models.SponsorModel;
+
+namespace DDDEastAnglia.Areas.Admin.Controllers
+{
+    public class SponsorController : Controller
+    {
+        private readonly ISponsorRepository sponsorRepository;
+        private readonly DefaultSponsorSorter _defaultSponsorSorter;
+
+        public SponsorController(ISponsorRepository sponsorRepository, DefaultSponsorSorter _defaultSponsorSorter)
```

Comment:
My concern is that the class is no longer open for extension by changing this to a concrete type. I'm not fussed exactly how we achieve that - delegates are fine, although I quite like the Strategy pattern approach because it allows you to put names on the concepts. It's very much an OO vs. Functional thing, though, and given C# straddles the two paradigms, there's no clear answer.

I tend to use delegates (Functional style) when the delegate will nearly always be a trivial lambda, and the Strategy pattern when the situation is more complex. The SponsorSorter concern may be still sufficiently complex to justify a full-on Strategy implementation.